### ISSUE TITLE
Question: How do I define an arbitrary URL

### ISSUE THREAD

**Issue by xspager** [sailorproject/sailor#112]
I'm wondering if it is possible to have a controller to respond to a URL like /books/2016/03/22/new_book, for what I checked it end up running the action index and passig parameters like ?2016=03&22=new_book

**Comment by m1cr0man** [comment]
Yes, the [new Regex I wrote](https://github.com/sailorproject/sailor/pull/111/commits/1ce71344419cd9a8794cb1e559f9ba98eb57b5eb) for matching friendly URLs on Nginx does what you would expect.

However, if you don't want to make every URL behave this way (as that config would cause) I suspect you would be stuck to rewriting the URL serverside in your controller.

**Comment by Etiene** [comment]
that would depend on the webserver you are using. If you are using apache, for example, you would have to edit the .htaccess file at the root dir of the app and change the regex there.

**Comment by Etiene** [comment]
ping! Did you succeed with this?

**Comment by xspager** [comment]
Oh, sorry for not answering earlier :( I just gave up using the format I wanted for the URL for now.

I have a problem with relying on the webserver to support more complicated URLs, if for example I wanted to go from Apache to Nginx I would need to rewrite the rules by hand. Would be nice if Sailor itself could handle the "complex" URLs or provided a tool to generate a config file or fragment to a particular webserver from a common format (Lua code/config?)

**Comment by Etiene** [comment]
This has yet to be implemented :x
